test/common/alerts_exist.go

```diff
@@ -0,0 +1,530 @@
+	var promApiCallOutput prometheusAPIResponse
+	err = json.Unmarshal([]byte(output), &promApiCallOutput)
+	if err != nil {
+		t.Fatal("Failed to unmarshal json:", err)
+	}
+	var rulesResult prometheusv1.RulesResult
+	err = json.Unmarshal([]byte(promApiCallOutput.Data), &rulesResult)
+	if err != nil {
+		t.Fatal("Failed to unmarshal json:", err)
+	}
+
+	// convert prometheus rule to PrometheusRule type
+	var actualRules []alertsTestRule
+	for _, group := range rulesResult.Groups {
+		ruleName := strings.Split(group.File, "/")
+		rule := alertsTestRule{
+			File: ruleName[len(ruleName)-1],
+		}
+		for _, promRule := range group.Rules {
+			switch v := promRule.(type) {
+			case prometheusv1.RecordingRule:
+				recRule := promRule.(prometheusv1.RecordingRule)
+				rule.Rules = append(rule.Rules, recRule.Name)
+			case prometheusv1.AlertingRule:
+				alertRule := promRule.(prometheusv1.AlertingRule)
+				rule.Rules = append(rule.Rules, alertRule.Name)
+			default:
+				fmt.Printf("unknown rule type %s", v)
+			}
+		}
+		actualRules = append(actualRules, rule)
+	}
+
+	// build up a reportMapping of missing or unexpected files
+	reportMapping := make(map[string]*alertsTestReport, 0)
+
+	// if an unexpected file is found, add it to the reportMapping
+	ruleDiff := ruleDifference(actualRules, expectedRules)
+	for _, rule := range ruleDiff {
+		reportMapping[rule.File] = &alertsTestReport{
+			AdditionalRules: rule.Rules,
+			Status:          fileAdditional,
+		}
+	}
+
+	// if an expected file is not found, add it to the reportMapping
+	ruleDiff = ruleDifference(expectedRules, actualRules)
+	for _, rule := range ruleDiff {
+		reportMapping[rule.File] = &alertsTestReport{
+			MissingRules: rule.Rules,
+			Status:       fileMissing,
+		}
+	}
+
+	// the file exists, do left and right diffs to ensure
+	// all rules exist and no unexpected rules are found
+	for _, actualRule := range actualRules {
+		for _, expectedRule := range expectedRules {
+			if actualRule.File == expectedRule.File {
+				reportMapping[actualRule.File] = buildReport(actualRule, expectedRule, reportMapping[actualRule.File])
+			}
+		}
+	}
+
+	// report the status
+	missingCount := 0
+	extraCount := 0
+	for k, v := range reportMapping {
+		if v.Status != fileCorrect {
+			fmt.Println("\nFile Name:", k)
+			fmt.Println("Missing Rules:", v.MissingRules)
+			fmt.Println("Additional Rules:", v.AdditionalRules)
+			fmt.Println("Status:", v.Status)
+		}
+		if v.Status == fileMissing || len(v.MissingRules) > 0 {
+			missingCount++
+		}
+		if v.Status == fileAdditional || len(v.AdditionalRules) > 0 {
+			extraCount++
+		}
+	}
+
+	if missingCount > 0 {
+		fmt.Println("\nMissing alerts were found from Prometheus. If the removal of these Alert rules was intentional, please update this test to remove them from the check. If the removal of these Alert rules was not intendended or you are not sure, please create a Jira & discuss with the monitoring team on how best to proceed")
```

Comment:
It would be good to print which alerts are missing or are extra

Reply:
Yep that happens here: https://github.com/integr8ly/integreatly-operator/pull/438/files#diff-fb2239bb04ca074b4768a268fef37e9aR421-R424

Reply:
Ok thanks, didn't notice it